### PR TITLE
Fix eigvals(:::SymTridiagonal) with longer off-diagonal vector

### LINKED ISSUE #40154: eigvals(::SymTridiagonal) errors if off-diagonal has wrong length
The following errors. I can prepare a fix but I don't understand why `SymTridiagonal(randn(5), randn(5))` works: I would expect it to require a length `4` off-diagonal vector. The docs don't clarify anything...

```julia
julia> A = SymTridiagonal(randn(5), randn(5))
5×5 SymTridiagonal{Float64, Vector{Float64}}:
 1.54728  1.29122     ⋅          ⋅          ⋅ 
 1.29122  1.7491     0.820245    ⋅          ⋅ 
  ⋅       0.820245  -0.933958   0.816364    ⋅ 
  ⋅        ⋅         0.816364  -0.2896    -0.552447
  ⋅        ⋅          ⋅        -0.552447   0.669905

julia> eigvals(A)
ERROR: DimensionMismatch("ev has length 5 but needs one less than dv's length, 5)")
Stacktrace:
 [1] stev!(job::Char, dv::Vector{Float64}, ev::Vector{Float64})
   @ LinearAlgebra.LAPACK ~/Projects/julia-1.6/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/lapack.jl:3774
 [2] eigvals!
   @ ~/Projects/julia-1.6/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/tridiag.jl:293 [inlined]
 [3] eigvals(A::SymTridiagonal{Float64, Vector{Float64}})
   @ LinearAlgebra ~/Projects/julia-1.6/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/tridiag.jl:294
 [4] top-level scope
   @ REPL[3]:1
```

### PR DESCRIPTION
**Pull request by dlfivefifty**
This resolves #40154

### CODE REVIEW

**Comment by dlfivefifty** [comment]
Note `eigen` was already fixed here: https://github.com/dlfivefifty/julia/commit/ad523d5684cd9bbb98b7b63b472c53ea2d200cc4

**Review by dkarrasch — APPROVED** [review]

**Comment by vtjnash** [comment]
Appears we cannot merge due to test failurs

File: stdlib/LinearAlgebra/src/lapack.jl [hunk 1]
```diff
@@ -3770,7 +3770,7 @@ for (stev, stebz, stegr, stein, elty) in
             require_one_based_indexing(dv, ev)
             chkstride1(dv, ev)
             n = length(dv)
-            if length(ev) != n - 1 && length(ev) != n
+            if length(ev) != n - 1 || length(ev) != n
```
Comment (dkarrasch) [inline stdlib/LinearAlgebra/src/lapack.jl:3773]: I think the `&&` was correct.

**Comment by dkarrasch** [comment]
The previous issue was that

https://github.com/JuliaLang/julia/blob/8218480f059b7d2ba3388646497b76759248dd86/stdlib/LinearAlgebra/test/lapack.jl#L410

no longer threw a `DimensionMismatch` error, so the 10 should be replaced by 11, or the test removed entirely.

File: stdlib/LinearAlgebra/src/lapack.jl (hunk 1 above)
Comment (dlfivefifty) [reply stdlib/LinearAlgebra/src/lapack.jl:3773]: Oh yeah sorry

**Pull request merged**